lua/apisix/core/request.lua

```diff
@@ -19,6 +22,27 @@ local function _headers(ctx)
 
     return headers
 end
+
+local function _validate_header_name(name)
+    local tname = type(name)
+    if tname ~= "string" then
+        return nil, str_fmt("invalid header name %q: got %s, " ..
+                "expected string", name, tname)
+    end
+
+    return name, nil
+end
+
+local function _validate_header_value(value)
+    local tvalue = type(value)
+    if tvalue ~= "string" and tvalue ~= "number" and tvalue ~= "boolean" then
```

Comment:
I think we can remove the type check

Reply:
Why?

Reply:
The header_value can take an array list of values, for example,
```
 ngx.req.set_header("Foo", {"a", "abc"})
```
will produce two new request headers:
```
 Foo: a
 Foo: abc
```